diff --git a/sites/all/modules/contrib/leaflet/leaflet.drupal.js b/sites/all/modules/contrib/leaflet/leaflet.drupal.js
index 33d3566..9250833 100644
--- a/sites/all/modules/contrib/leaflet/leaflet.drupal.js
+++ b/sites/all/modules/contrib/leaflet/leaflet.drupal.js
@@ -107,7 +107,7 @@
             lFeature = Drupal.leaflet.create_multipoly(feature);
             break;
           case 'json':
-            lFeature = Drupal.leaflet.create_json(feature.json)
+            lFeature = Drupal.leaflet.create_json(feature.json);
             break;
         }
 
@@ -147,14 +147,15 @@
       // layers served from TileStream need this correction in the y coordinates
       // TODO: Need to explore this more and find a more elegant solution
       if (layer.type == 'tilestream') {
-        map_layer.getTileUrl = function (tilePoint, zoom) {
-          var subdomains = this.options.subdomains,
-            s = this.options.subdomains[(tilePoint.x + tilePoint.y) % subdomains.length];
-
-          return this._url
-            .replace('{z}', zoom)
-            .replace('{x}', tilePoint.x)
-            .replace('{y}', Math.pow(2, zoom) - tilePoint.y - 1);
+        map_layer.getTileUrl = function (tilePoint) {
+          this._adjustTilePoint(tilePoint);
+          var zoom = this._getZoomForUrl();
+          return L.Util.template(this._url, L.Util.extend({
+            s: this._getSubdomain(tilePoint),
+            z: zoom,
+            x: tilePoint.x,
+            y: Math.pow(2, zoom) - tilePoint.y - 1
+          }, this.options));
         }
       }
       return map_layer;
@@ -233,7 +234,11 @@
     },
 
     create_json: function(json) {
-      lJSON = new L.GeoJSON();
+      lJSON = new L.GeoJSON(json, {
+        onEachFeature: function(feature, layer) {
+          layer.bindPopup(feature.properties.popup);
+        }
+      });
 
       lJSON.on('featureparse', function (e) {
         e.layer.bindPopup(e.properties.popup);
@@ -253,7 +258,6 @@
         }
       });
 
-      lJSON.addGeoJSON(json);
       return lJSON;
     },
 
